<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Robot Test</string>
  </property>
  <property name="statusTip">
   <string/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="4" column="0">
     <layout class="QHBoxLayout" name="link_layout">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="lbl_mode">
        <property name="text">
         <string>Link    </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spn_link_index">
        <property name="wrapping">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spn_driver_index">
        <property name="visible">
         <bool>false</bool>
        </property>
        <property name="wrapping">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cb_link"/>
      </item>
      <item>
       <widget class="QComboBox" name="cb_driver">
        <property name="visible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Angle</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="spn_link">
        <property name="accelerated">
         <bool>true</bool>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.030000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="spn_driver">
        <property name="visible">
         <bool>false</bool>
        </property>
        <property name="accelerated">
         <bool>true</bool>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="minimum">
         <double>-99.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.030000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lbl_driver_info">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>info</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lbl_link_info">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>info</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QRobotTestBackend" name="displaywidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>200</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="toolTip">
       <string>free mode</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <widget class="QCheckBox" name="chk_com">
        <property name="statusTip">
         <string>draw COM</string>
        </property>
        <property name="text">
         <string>COM</string>
        </property>
       </widget>
      </item>
      <item row="0" column="7">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="5">
       <widget class="QCheckBox" name="chk_expanded">
        <property name="statusTip">
         <string>draw expanded</string>
        </property>
        <property name="text">
         <string>expanded</string>
        </property>
        <property name="tristate">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <widget class="QCheckBox" name="chk_collisions">
        <property name="statusTip">
         <string>draw collision map</string>
        </property>
        <property name="text">
         <string>collisions</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QCheckBox" name="chk_frame">
        <property name="statusTip">
         <string>draw frames</string>
        </property>
        <property name="text">
         <string>frame</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QCheckBox" name="chk_bboxes">
        <property name="statusTip">
         <string>draw bounding boxes</string>
        </property>
        <property name="text">
         <string>bboxes</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="chk_geometry">
        <property name="statusTip">
         <string>draw the robot geometry</string>
        </property>
        <property name="text">
         <string>geometry</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="6">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QRadioButton" name="radioButton">
          <property name="text">
           <string>Link</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_2">
          <property name="text">
           <string>Driver</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="sld_link">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="sld_driver">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_ik">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="icons.qrc">
            <normaloff>:/Images/Images/IK.gif</normaloff>:/Images/Images/IK.gif</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Print Collisions</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
   <zorder>displaywidget</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_File"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="statusTip">
    <string>test123</string>
   </property>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="statusTip">
    <string>ctrl-o</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_State">
   <property name="text">
    <string>Save State</string>
   </property>
   <property name="toolTip">
    <string>ctrl-s</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoad_MultiPath">
   <property name="text">
    <string>Load MultiPath</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionSaveView">
   <property name="text">
    <string>Save View</string>
   </property>
   <property name="toolTip">
    <string>v</string>
   </property>
  </action>
  <action name="actionRevertView">
   <property name="text">
    <string>Revert</string>
   </property>
   <property name="toolTip">
    <string>V</string>
   </property>
  </action>
  <action name="actionDefaultView">
   <property name="text">
    <string>Default</string>
   </property>
  </action>
  <action name="actionDrivers">
   <property name="text">
    <string>Drivers</string>
   </property>
   <property name="toolTip">
    <string>ctrl-d</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionSensor_Plot">
   <property name="text">
    <string>Sensor Plot</string>
   </property>
   <property name="toolTip">
    <string>ctrl-p</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;ctrl-r&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionSave_State_As">
   <property name="text">
    <string>Save State As</string>
   </property>
   <property name="toolTip">
    <string>ctrl-shift-s</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionRecordFile">
   <property name="text">
    <string>Change File</string>
   </property>
  </action>
  <action name="actionMultiple_Parts">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Multiple Parts</string>
   </property>
   <property name="statusTip">
    <string>Does not automatically encode when a recording ends.  Can record 2 parts consecutively into 1 video.</string>
   </property>
  </action>
  <action name="action640x480">
   <property name="text">
    <string>640x480</string>
   </property>
  </action>
  <action name="action800x600">
   <property name="text">
    <string>800x600</string>
   </property>
  </action>
  <action name="action1920x1080_HD">
   <property name="text">
    <string>1920x1080 (HD)</string>
   </property>
  </action>
  <action name="action1280x768">
   <property name="text">
    <string>1280x768</string>
   </property>
  </action>
  <action name="actionResizable">
   <property name="text">
    <string>Resizable</string>
   </property>
  </action>
  <action name="actionEncode">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Encode</string>
   </property>
  </action>
  <action name="actionSend_Command">
   <property name="text">
    <string>Send Command</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionOpen_File">
   <property name="text">
    <string>Open File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QRobotTestBackend</class>
   <extends>QWidget</extends>
   <header>qrobottestbackend.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>chk_geometry</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetGeometry(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>495</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>569</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_com</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetCOM(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>246</x>
     <y>495</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>571</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_frame</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetFrame(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>571</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_bboxes</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetBboxes(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>407</x>
     <y>543</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_expanded</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetExpanded(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>510</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>570</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_collisions</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetCollisions(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>607</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>624</x>
     <y>552</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cb_link</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetLink(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>573</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_ik</sender>
   <signal>clicked(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetIK(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>530</y>
    </hint>
    <hint type="destinationlabel">
     <x>28</x>
     <y>486</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spn_link_index</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cb_link</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>261</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spn_link_index</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetLink(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>41</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cb_link</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>spn_link_index</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>LinkMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>520</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>DriverMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>517</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>462</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spn_driver_index</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetDriver(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>25</x>
     <y>549</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spn_driver_index</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cb_driver</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cb_driver</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>spn_driver_index</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cb_driver</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SetDriver(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>567</y>
    </hint>
    <hint type="destinationlabel">
     <x>541</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sld_link</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SliderLinkAngle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>524</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>490</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sld_driver</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SliderDriverAngle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>525</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>480</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spn_link</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>UpdateLinkSlider(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>436</x>
     <y>557</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>548</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spn_driver</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>UpdateDriverSlider(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>518</x>
     <y>557</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>574</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>PrintCollisions()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>628</x>
     <y>508</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_File</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>LoadFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>ShowHelp()</slot>
  <slot>ShowAbout()</slot>
  <slot>SetGeometry(bool)</slot>
  <slot>SetCOM(bool)</slot>
  <slot>SetFrame(bool)</slot>
  <slot>SetBboxes(bool)</slot>
  <slot>SetExpanded(bool)</slot>
  <slot>SetCollisions(bool)</slot>
  <slot>SetLink(int)</slot>
  <slot>SetDriver(int)</slot>
  <slot>SetDriverValue(double)</slot>
  <slot>SetIK(bool)</slot>
  <slot>LinkMode()</slot>
  <slot>DriverMode()</slot>
  <slot>SliderLinkAngle(int)</slot>
  <slot>SliderDriverAngle(int)</slot>
  <slot>UpdateLinkSlider(double)</slot>
  <slot>UpdateDriverSlider(double)</slot>
  <slot>PrintCollisions()</slot>
  <slot>LoadFile()</slot>
 </slots>
</ui>
